.github/workflows/build.yml

```diff
@@ -22,7 +22,7 @@ jobs:
         uses: actions/setup-java@v4
         with:
           distribution: 'adopt'
-          java-version: 8
+          java-version: 21
           cache: 'maven'
           cache-dependency-path: |
```

Comment:
We need a section on the README that this is a breaking change to spark 4 and will not be compat with spark3   runtimes